contrib/win32/win32compat/pwd.c

```diff
@@ -237,7 +237,7 @@ get_passwd(const wchar_t * user_utf16, PSID sid)
 	}
 
 	/* If standard local user name, just use name without decoration */
-	if ((_wcsicmp(domain_name, computer_name) == 0))
+	if ((_wcsicmp(domain_name, computer_name) == 0) && (_wcsicmp(computer_name, user_name) != 0))
```

Comment:
With this change, I don't think my original PR is not required as we always resolve the user to "<user_name aka machine_name>\<user_name aka machine_name>"

Reply:
Yes, correct. I just tested it on sshd and it works fine with just this one-line change.